src/deadline/job_attachments/api/manifest.py

```diff
@@ -0,0 +1,140 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+import datetime
+from pathlib import Path
+from typing import List, Optional, Tuple
+
+
+from deadline.client.api._job_attachment import _hash_attachments
+from deadline.client.cli._common import _ProgressBarCallbackManager
+from deadline.client.cli._groups.click_logger import ClickLogger
+from deadline.job_attachments._diff import compare_manifest
+from deadline.job_attachments._glob import _process_glob_inputs, _glob_paths
+from deadline.job_attachments.asset_manifests.base_manifest import (
+    BaseManifestPath,
+)
+from deadline.job_attachments.asset_manifests.decode import decode_manifest
+from deadline.job_attachments.models import (
+    FileStatus,
+    GlobConfig,
+    JobAttachmentS3Settings,
+    ManifestSnapshot,
+    default_glob_all,
+)
+from deadline.job_attachments.upload import S3AssetManager
+
+"""
+APIs here should be business logic only. It should perform one thing, and one thing well. 
+It should use basic primitives like S3 upload, download, boto3. 
+These APIs should be boto3 session agnostic and given which underlying credentials are used.
+"""
+
+
+def _manifest_snapshot(
+    root: str,
+    destination: str,
+    name: str,
+    include: Optional[List[str]],
+    exclude: Optional[List[str]],
+    include_exclude_config: Optional[str],
+    diff: Optional[str] = None,
+    logger: ClickLogger = ClickLogger(False),
+) -> Optional[ManifestSnapshot]:
+
+    # Get all files in the root.
+    glob_config: GlobConfig
+    if include or exclude:
+        include = include if include is not None else default_glob_all()
+        exclude = exclude if exclude is not None else []
+        glob_config = GlobConfig(include_glob=include, exclude_glob=exclude)
+    elif include_exclude_config:
+        glob_config = _process_glob_inputs(include_exclude_config)
+    else:
+        # Default, include all.
+        glob_config = GlobConfig()
+
+    inputs = _glob_paths(root, include=glob_config.include_glob, exclude=glob_config.exclude_glob)
+
+    # Placeholder Asset Manager
+    asset_manager = S3AssetManager(
+        farm_id=" ", queue_id=" ", job_attachment_settings=JobAttachmentS3Settings(" ", " ")
+    )
```

Comment:
Hmmmm yeah this particular line came from the prior branch. Let me look into it.

Reply:
I think we need to refactor this function as a separate utility altogether. I'll take that as a follow up refactor because there are 40 references from code search to this method.